### DISCUSSION TITLE
matplotlib.pyplot.show() not creating windows?

### DISCUSSION THREAD

**Discussion by ghylander** [brentvollebregt/auto-py-to-exe#330] [Q&A]
I looked a bit around this repo and PyInstaller's, but I couldn't find how to write the code so that figures are shown, or whether or not it's possible to do so.

My python app requires some user input through console, then displays some pyplot figures through pyplot.show()
How to achieve this?

**Comment by ghylander** [comment]
I tried to include both the `--console` and `--windowed` options but it didn't work

**Reply by ghylander** [reply]
Also tried the `from multiprocessing import freeze_support` and `freeze_support()` lines, jsut in case there was a problem with matplotlib not correctly blocking the process to `show()` the figures

**Reply by ghylander** [reply]
Also, note that there's no Exception being raised, or any other error for that matter.
I included some prints around the code when the figures are created, updated and finally showed, and they print fine as well

**Reply by ghylander** [reply]
I also added the `libiomp5md.dll` file (as mentioned [here](https://github.com/pyinstaller/pyinstaller/issues/4968)), both as `--add-binary` command and including it in the 'exe' directory

**Reply by brentvollebregt** [reply]
> Also, note that there's no Exception being raised, or any other error for that matter.

Did you follow the debugging guideline? If there are no exceptions, it sounds like the application is doing what it's told.

**Reply by ghylander** [reply]
> Did you follow the debugging guideline? If there are no exceptions, it sounds like the application is doing what it's told.

Yeah, I did
The application should open various pyplot windows, one after another. But it doesn't
Additionally, I tried saving the figures with `pyplot.savefig()`, but the figures are not saving to disk either

**Reply by brentvollebregt** [reply]
If you run the application from a terminal, does it simply complete with an exit status code of 0?

**Reply by ghylander** [reply]
Since it seemed more of a problem with PyInstaller than auto-py-to-exe, I opened a [discussion there](https://github.com/orgs/pyinstaller/discussions/7286) and was able to determine the cause of the issue.
It turns out, a library I use in my program (mpld3) internally calls `matplotlib.use('Agg')`. This caused PyInstaller to only select the Agg matplotlib backend. The parts of my program which used a different backend did not work then.
I called `matplotlib.use('Qt5Agg')` in my code, which resulted in PyInstaller to selecting both the Agg and Qt5Agg backends and now everything works fine.

Thanks for the replies though!
